### ISSUE TITLE
[Enhancement]: Add Whisper support

### ISSUE THREAD

**Issue by apiweb** [advplyr/audiobookshelf#1723]
### Describe the feature/enhancement

Hi there!

I've been using AudioBookShelf for a while now, and I love the platform. I was thinking about how it could be improved, and I had an idea that I wanted to share with you all.

I think it would be great if AudioBookShelf could integrate with [Whisper](https://github.com/openai/whisper) speech-to-text model to automatically generate subtitles for audiobooks. This could be an external tool like Tone and ffmpeg that the user could enable or disable as needed.

With Whisper, it would be possible to transcribe speech across dozens of languages and even handle poor audio quality or excessive background noise. It would make it easier for people who are hard of hearing or have difficulty understanding accents to enjoy audiobooks.

Here are some tips on how to integrate this feature into the AudioBookShelf flow:

- Use the metadata language tag to automatically set the Whisper language.
- Automatically save the srt using the [Title Folder Naming](https://www.audiobookshelf.org/docs#book-title-folder-naming) structure.

I hope you will consider this suggestion for future updates to AudioBookShelf. Let me know if you have any questions or concerns.

Thank you for all your hard work making AudioBookShelf a great platform!

**Comment by advplyr** [comment]
Maybe we first add support for an `srt` subtitle file.

The opposite of this was also requested for ebooks #601

**Comment by advplyr** [comment]
There is also this issue https://github.com/advplyr/audiobookshelf/issues/189

**Comment by damajor** [comment]
I tested Whisper on my setup and results are kind of good but far from perfect.

Using base model it took me around 4 minutes to transcribe 1 hour audiobook.
Using large model it took me a bit more than 1 hour to transcribe 1 hour audiobook.

Those tests were done on 5950X with 12 parallel threads (no GPU involved).

**Comment by Ed1ks** [comment]
👍thumbs up. Would be very nice for self recorded audiofiles.

**Comment by rounakdatta** [comment]
Should we consider working on this? I can volunteer to start contributing.
Currently [Snipd](https://www.snipd.com/) is the only app which gets AI-generated subtitles UX perfectly right, and this could be our chance to make Audiobookshelf the FOSS alternative to it.

**Comment by advplyr** [comment]
Yeah I'm interested in this but I can't put much attention towards it now. We were talking about it in Discord the other day. If anyone wants to start putting something together or setup a proof of concept that would be great. We can chat about it in Discord

**Comment by yuchen-lea** [comment]
> Maybe we first add support for an `srt` subtitle file.也许我们首先添加对 `srt` 字幕文件的支持。
> 
> The opposite of this was also requested for ebooks #601电子书 #601 也提出了相反的要求

@advplyr I agree with what you said about adding support for SRT subtitle files first. I have now used Whisper to generate corresponding subtitles for my local podcasts. On my computer, I can search and view them. Displaying subtitles while playing on the ABS mobile app is the final piece of the jigsaw.

I think these two things can share the same UI: LRC files #817 (external LRC files with the same name as the audio file or ID3 information embedded in the audio file) and SRT files #2257 (external SRT files with the same name as the audio file).

**Comment by iamhenry** [comment]
> Should we consider working on this? I can volunteer to start contributing. Currently [Snipd](https://www.snipd.com/) is the only app which gets AI-generated subtitles UX perfectly right, and this could be our chance to make Audiobookshelf the FOSS alternative to it.

this is exactly what would be great with ABS. I have the paid feature for Snipd and now it's hard to take notes without it using audiobooks.

**Comment by turnercore** [comment]
> just FYI there are several implementations of whisper specifically tailored to subtitle generation. This one for example https://github.com/jianfch/stable-ts can not only generate srt, but also ssa/ass karaoke style subs [meaning that the current spoken word is highlighted] bringing us even closer to snipd. From my experience base and small models are enough with it.

Thanks for the heads up, hopefully I can get some time to work on a PR for this type of thing. I haven't contributed yet though so I imagine it will take me a bit to get familiar with the code base and what needs to be updated for this kind of feature.

**Comment by iamhenry** [comment]
@turnercore that would be awesome!
